MessageFormatterMethod does not detect/map SimpleNumber causing IllegalArgumentException

### Before reporting an issue

- [x] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

login/ui

### Describe the bug

When using a Java MessageFormat with Choice format in .properties files like: 

foo="{0,choice,0#foo|1#bar|1<{0} foobar}"

and calling it within a FTL like this:

${msg("foo" , 3)}

causes an IllegalArgumentException: Cannot format given Object as a Number.

This is because the MessageFormatterMethod does not map SimpleNumbers to Numbers before formatting it with 
java.text.Format::format


### Version

main

### Regression

- [ ] The issue is a regression

### Expected behavior

When using the choice format foo="{0,choice,0#foo|1#bar|1<{0} foobar}" and calling it with ${msg("foo" , 3)} i want the correct formatted String 

### Actual behavior

${msg("foo" , 3)}

causes an IllegalArgumentException: Cannot format given Object as a Number.

### How to Reproduce?

Add the property  foo="{0,choice,0#foo|1#bar|1<{0} foobar}" to a .properties file e.g. (messages_en.properties)

Call this property from a FTL file e.g. (login.ftl) ${msg("foo" , 3)}

Build keycloak and navigate to the ftl in use.

Result:
Caused by: java.lang.IllegalArgumentException: Cannot format given Object as a Number
        at java.base/java.text.NumberFormat.format(NumberFormat.java:283)
        at java.base/java.text.Format.format(Format.java:159)
        at java.base/java.text.MessageFormat.subformat(MessageFormat.java:1318)
        at java.base/java.text.MessageFormat.format(MessageFormat.java:912)
        at java.base/java.text.Format.format(Format.java:159)
        at org.keycloak.theme.beans.MessageFormatterMethod.exec(MessageFormatterMethod.java:62)
        at freemarker.core.MethodCall._eval(MethodCall.java:62)
        at freemarker.core.Expression.eval(Expression.java:101)
        at freemarker.core.DollarVariable.calculateInterpolatedStringOrMarkup(DollarVariable.java:104)
        at freemarker.core.DollarVariable.accept(DollarVariable.java:63)
        at freemarker.core.Environment.visit(Environment.java:335)
        at freemarker.core.Environment.visit(Environment.java:377)
        at freemarker.core.Environment.invokeNestedContent(Environment.java:621)
        at freemarker.core.BodyInstruction.accept(BodyInstruction.java:60)
        at freemarker.core.Environment.visit(Environment.java:335)
        at freemarker.core.Environment.visit(Environment.java:341)
        at freemarker.core.Environment.visit(Environment.java:377)
        at freemarker.core.Environment.invokeMacroOrFunctionCommonPart(Environment.java:877)
        at freemarker.core.Environment.invokeMacro(Environment.java:813)
        at freemarker.core.UnifiedCall.accept(UnifiedCall.java:84)
        at freemarker.core.Environment.visit(Environment.java:335)
        at freemarker.core.Environment.visit(Environment.java:341)
        at freemarker.core.Environment.process(Environment.java:314)
        at freemarker.template.Template.process(Template.java:383)
        at org.keycloak.theme.freemarker.DefaultFreeMarkerProvider.processTemplate(DefaultFreeMarkerProvider.java:49)
        ... 26 more



### Anything else?

_No response_

--- a/services/src/main/java/org/keycloak/theme/beans/MessageFormatterMethod.java
+++ b/services/src/main/java/org/keycloak/theme/beans/MessageFormatterMethod.java
@@ -19,6 +19,7 @@ package org.keycloak.theme.beans;
 
 import static java.util.Optional.ofNullable;
 
+import freemarker.template.SimpleNumber;
 import freemarker.template.SimpleScalar;
 import freemarker.template.TemplateMethodModelEx;
 import freemarker.template.TemplateModelException;
@@ -67,11 +68,14 @@ public class MessageFormatterMethod implements TemplateMethodModelEx {
     private List<Object> resolve(List<Object> list) {
         ArrayList<Object> result = new ArrayList<>();
         for (Object item: list) {
-            if (item instanceof SimpleScalar) {
-                item = ((SimpleScalar) item).getAsString();
+            if (item instanceof SimpleScalar scalar) {
+                item = scalar.getAsString();
+            } else if (item instanceof SimpleNumber number) {
+                item = number.getAsNumber();
             }
-            if (item instanceof String) {
-                result.add(TemplatingUtil.resolveVariables((String) item, messages));
+
+            if (item instanceof String string) {
+                result.add(TemplatingUtil.resolveVariables(string, messages));
             } else {
                 result.add(item);
             }

--- a/services/src/test/java/org/keycloak/theme/beans/MessageFormatterMethodTest.java
+++ b/services/src/test/java/org/keycloak/theme/beans/MessageFormatterMethodTest.java
@@ -17,6 +17,8 @@
 
 package org.keycloak.theme.beans;
 
+import freemarker.template.SimpleNumber;
+import freemarker.template.SimpleScalar;
 import freemarker.template.TemplateModelException;
 import org.junit.Assert;
 import org.junit.Test;
@@ -40,6 +42,7 @@ public class MessageFormatterMethodTest {
         properties.setProperty("backToClient", "Back to {0}");
         properties.setProperty("client_admin-console", "Admin Console");
         properties.setProperty("realm_example-realm", "Example Realm");
+        properties.setProperty("key", "foo {0,choice,0#foo|1#bar|1<{0} foobar} bar");
 
 
         MessageFormatterMethod fmt = new MessageFormatterMethod(locale, properties);
@@ -52,5 +55,15 @@ public class MessageFormatterMethodTest {
 
         msg = (String) fmt.exec(Arrays.asList("backToClient", "client '${client_admin-console}' from '${realm_example-realm}'."));
         Assert.assertEquals("Back to client 'Admin Console' from 'Example Realm'.", msg);
+
+        msg = (String) fmt.exec(Arrays.asList(new SimpleScalar("key"),new SimpleNumber(0)));
+        Assert.assertEquals("foo foo bar", msg);
+
+        msg = (String) fmt.exec(Arrays.asList(new SimpleScalar("key"),new SimpleNumber(1)));
+        Assert.assertEquals("foo bar bar", msg);
+
+        msg = (String) fmt.exec(Arrays.asList(new SimpleScalar("key"),new SimpleNumber(2)));
+        Assert.assertEquals("foo 2 foobar bar", msg);
     }
+
 }

